src/freenet/client/async/ClientLayerPersister.java

```diff
@@ -230,16 +230,16 @@ private boolean innerSetFilesAndLoad(boolean noSerialize, File dir, String baseN
         }
         boolean failedSerialize = false;
         PartialLoad loaded = new PartialLoad();
-        if(clientDatExists) {
+        if (clientDatExists) {
             innerLoad(loaded, makeBucket(dir, baseName, false, null), noSerialize, context, requestStarters, random);
         }
-        if(clientDatCryptExists && loaded.needsMore()) {
+        if (clientDatCryptExists) {
```

Comment:
If a corrupt `client.dat.crypt` exists, even when encryption is disabled, it will now always mess up the loaded `client.dat`.